src/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -120,6 +120,9 @@ public function load(array $configs, ContainerBuilder $container): void
         $patchFormats = $this->getFormats($config['patch_formats']);
         $errorFormats = $this->getFormats($config['error_formats']);
         $docsFormats = $this->getFormats($config['docs_formats']);
+        $jsonSchemaFormats = $config['jsonschema_formats']
+            ? $this->getFormats($config['jsonschema_formats'])
+            : array_filter($formats, static fn ($format) => !\in_array($format, ['html', 'multipart', 'xml'], true), \ARRAY_FILTER_USE_KEY);
```

Comment:
Agree, let's try this (pseudo code, use array_filter if you prefer I don't mind):

```suggestion
        $jsonSchemaFormats = $config['jsonschema_formats'];
        
        if (!$jsonSchemaFormats) {
            foreach (array_keys($formats) as $f) {
              if (str_starts_with('json', $f) { 
                $jsonSchemaFormats[] = $f;
              }
            }
        }
```

We don't need mime types for this so let's get rid on the `getFormats`, and if not defined by the user we want this to be defined for all the supported json formats.